### ISSUE TITLE
Matrix multiply is not commutative

### ISSUE THREAD

**Issue by Elektraglide** [laurentlb/shader-minifier#472]
you cannot swap order of multiplies for matrices; it changes the meaning.
I have vec4 v = mat33 * (vec3 * float)  which is transformed into:
vec4 v = (vec3 * float) * mat33

Is rewriter.fs:240 the problem?

**Comment by laurentlb** [comment]
Which version of Shader Minifier are you using? I think we fixed some instances of the bug.

Can you share the line of code that triggers the bug?

**Comment by eldritchconundrum** [comment]
My guess is, it's this rewrite: `x*(y*z) -> y*z*x`
How to reproduce:
`mat2 f(){ return mat3x2(1) * (mat4x3(1) * mat2x4(1)); }`
becomes
`mat2 f(){return mat4x3(1)*mat2x4(1)*mat3x2(1);}`

**Comment by eldritchconundrum** [comment]
No, but commenting out lines 198-200 of rewriter.fs should fix it

**Comment by Elektraglide** [comment]
dumb question:  where do I find the version I'm building?   I built it on another laptop over the summer and can't see anything that shows version.  I thought shader_minifier.exe -v  might give some blurb..

**Comment by Elektraglide** [comment]
yep, that works.  cheers

**Comment by laurentlb** [comment]
Good point, we need a `--version` flag. :)

The version number is included in (some of) the generated outputs, but you can also run `.\shader_minifier.exe --help` and look at the first line.
